### PR TITLE
[Windows] Allow early use of DeviceDisplay.Current.MainDisplayInfo

### LINKED ISSUE #6976: DeviceDisplay.Current.MainDisplayInfo not available on windows
### Description

DeviceDisplay.Current.MainDisplayInfo not available on windows

### Steps to Reproduce

Create a Maui Application and add the following in AppShell Constructor:
    public AppShell()
    {
        InitializeComponent();
        var di = DeviceDisplay.Current.MainDisplayInfo;
    }

The displayInfo does not provide any information about the display

### Version with bug

Release Candidate 2 (current)

### Last version that worked well

Unknown/Other

### Affected platforms

Windows

### Affected platform versions

Windows 11 Pro 22000.408

### Did you find any workaround?

no

### Relevant log output

_No response_

### PR DESCRIPTION
**Pull request by Foda**
### Description of Change

Currently (on Windows), `DeviceDisplay.Current.MainDisplayInfo` doesn't get set until the first window is activated/shown on the screen since it uses the active window to gather this info.

This PR fixes that by setting the active window as soon as possible using the `OnPlatformWindowSubclassed` lifecycle event. This lifecycle event happens after we have a valid `HWND` but before the window is "technically" active. I don't think this will cause any issues (???).

As a result, Windows users can now do things like override `Application.CreateWindow` to set the initial window position and size before it's shown without needing to pinvoke:
```csharp
class App : Application
{
    protected override Window CreateWindow(IActivationState activationState)
    {
        Window window = new Window(new NavigationPage(new MainPage()));

        var disp = DeviceDisplay.Current.MainDisplayInfo;
        var newWidth = disp.Width / 2;
        var newHeight = disp.Height / 2;

        // center the window
        window.X = (disp.Width / disp.Density - newWidth) / 2;
        window.Y = (disp.Height / disp.Density - newHeight) / 2;

        return window;
    }
}
```

### Issues Fixed
Fixes #6976

### CODE REVIEW

**Review by jsuarezruiz — APPROVED** [review]

**Review by jfversluis — APPROVED** [review]
I think a lot of people will be happy with this one, thank you!

File: src/Essentials/test/DeviceTests/Tests/Windows/DeviceDisplay_Windows_Tests.cs [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+
+				Assert.NotNull(testWindowStateManager.GetPlatformInitializedWindow());
+				Assert.NotNull(testWindowStateManager.GetActiveWindow());
+				Assert.True(metrics.Width > 0);
+				Assert.True(metrics.Height > 0);
+				Assert.True(metrics.Density > 0);
+			});
+		}
+
+		private class TestWindowStateManagerImplementation : IWindowStateManager
```
Comment (rmarinho) [inline src/Essentials/test/DeviceTests/Tests/Windows/DeviceDisplay_Windows_Tests.cs:38]:
```suggestion
		class TestWindowStateManagerImplementation : IWindowStateManager
```

**Comment by rmarinho** [comment]
Can we name the files for test like the other ones, no underscores please.

**Review by rmarinho — APPROVED** [review]

**Comment by kt215** [comment]
How will I know which maui-windows workload version this fix will be included in? The latest version is 7.0.81/7.0.100

**Comment by gerneio** [comment]
Would also like to know what version this fix will be applied to? Is this being saved for .NET 8? Hopefully will be delivered sooner.  Can confirm that it's still an issue in the latest maui-windows workload version.

**Comment by mattleibow** [comment]
I wonder if we should revert this actually and take a different approach. If there is no main window yet, then we can just use the main monitor on the system?

**Comment by dinther** [comment]
It's day two of me exploring .NET Maui  ...   And I run into fundamental stuff like this!

The workaround solution offered here https://github.com/dotnet/maui/issues/6976#issuecomment-1528006119
is of no use as it assumes a displayDensity of 1

When will the fix be released?

**Pull request merged**